component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -115,6 +118,37 @@ public BuilderImpl(final Schema providedSchema) {
             this.providedSchema = providedSchema;
         }
 
+        @Override
+        public Object getValue(final String name) {
+            return this.values.get(name);
+        }
+
+        @Override
+        public Builder with(final Entry entry, final Object value) {
```

Comment:
Why not... but same remark as above.

Reply:
May simplify client code like : "if (entry.getType() ... && value instanceof) ..."; only constraint is that type control become dynamic (like for nullable).